src/EnergyPlus/ScheduleManager.cc

```diff
@@ -4491,6 +4491,78 @@ namespace ScheduleManager {
+        // finds the minimum and maximum for a specific set of day types for a given schedule
+        Real64 MinValue = Constant::BigNumber;
+        Real64 MaxValue = -Constant::BigNumber;
+        //                                                           Sun    Mon   Tues  Wed    Thur  Fri   Sat    Hol   Summer Winter Cust1  Cust2
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterWkDy = {false, true, true, true, true, true, false, false, false, false, false, false};
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterWeHo = {true, false, false, false, false, false, true, true, false, false, false, false};
+        //  Sun    Mon    Tues   Wed    Thur   Fri    Sat    Hol    Summer Winter Cust1  Cust2
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterSumDsDy = {
+            false, false, false, false, false, false, false, false, true, false, false, false};
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterWinDsDy = {
+            false, false, false, false, false, false, false, false, false, true, false, false};
+        constexpr std::array<bool, maxDayTypes> dayTypeFilterNone = {
+            false, false, false, false, false, false, false, false, false, false, false, false};
+        if (ScheduleIndex > 0 && ScheduleIndex <= state.dataScheduleMgr->NumSchedules) {
+            int curDayTypeGroup = static_cast<int>(days);
+            auto &curSch = state.dataScheduleMgr->Schedule(ScheduleIndex);
+            if (!curSch.MaxMinByDayTypeSet[curDayTypeGroup]) {
+                std::array<bool, maxDayTypes> dayTypeFilter;
+                switch (days) {
+                case DayTypeGroup::Weekday:
+                    dayTypeFilter = dayTypeFilterWkDy;
+                    break;
+                case DayTypeGroup::WeekEndHoliday:
+                    dayTypeFilter = dayTypeFilterWeHo;
+                    break;
+                case DayTypeGroup::SummerDesignDay:
+                    dayTypeFilter = dayTypeFilterSumDsDy;
+                    break;
+                case DayTypeGroup::WinterDesignDay:
+                    dayTypeFilter = dayTypeFilterWinDsDy;
+                    break;
+                default:
+                    dayTypeFilter = dayTypeFilterNone;
+                    break;
+                }
+                for (int iDayOfYear = 1; iDayOfYear <= 366; ++iDayOfYear) {
+                    int WkSch = curSch.WeekSchedulePointer(iDayOfYear);
+                    auto &weekSch = state.dataScheduleMgr->WeekSchedule(WkSch);
+                    for (int jType = 1; jType <= maxDayTypes; ++jType) {
+                        if (dayTypeFilter[jType - 1]) {
+                            auto &daySch = state.dataScheduleMgr->DaySchedule(weekSch.DaySchedulePointer(jType));
+                            MinValue = min(MinValue, minval(daySch.TSValue));
+                            MaxValue = max(MaxValue, maxval(daySch.TSValue));
+                        }
```

Comment:
Rather than cycle through the entire schedule for every daytype group, why not cycle through once and store the max/min for each day type first? Then cycle through those to sort into the DayTypeGroups.

Reply:
@mjwitte I did some optimization of this loop that should greatly improve it's performance although it is not exactly using the approach you suggested.